textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexToGramQueryTranslator.java

```diff
@@ -44,16 +44,15 @@ public static GramBooleanQuery translate(String regex, int gramLength)
             throws com.google.re2j.PatternSyntaxException {
 
         GramBooleanQuery result = translateUnsimplified(regex, gramLength);
```

Comment:
Since we already get the unsimplified query during the process, there's no need for the `translateUnsimplified` function. Please merge the `translateUnsimplified` function into this main `translate` function.